feature/experimental/policy/policy_base/base_vrf_selection_test.go

```diff
@@ -0,0 +1,393 @@
+	ipv4FlowVlan20 := ate.Traffic().NewFlow("Ipv4Vlan20").WithSrcEndpoints(i1).WithDstEndpoints(i3).WithHeaders(ethHeader, ipv4Header).WithFrameSize(512).WithFrameRatePct(5)
+	ipv6FlowVlan20 := ate.Traffic().NewFlow("Ipv6Vlan20").WithSrcEndpoints(i1).WithDstEndpoints(i3).WithHeaders(ethHeader, ipv6Header).WithFrameSize(512).WithFrameRatePct(5)
+	ipipFlowVlan20 := ate.Traffic().NewFlow("IpipVlan20").WithSrcEndpoints(i1).WithDstEndpoints(i3).WithHeaders(ethHeader, ipipHeader, ipv4Header).WithFrameSize(512).WithFrameRatePct(5)
+	ipipDscp46FlowVlan20 := ate.Traffic().NewFlow("IpipDscp46Vlan20").WithSrcEndpoints(i1).WithDstEndpoints(i3).WithHeaders(ethHeader, ipipDscp46Header, ipv4Header).WithFrameSize(512).WithFrameRatePct(5)
+	ipipDscp42FlowVlan20 := ate.Traffic().NewFlow("IpipDscp42Vlan20").WithSrcEndpoints(i1).WithDstEndpoints(i3).WithHeaders(ethHeader, ipipDscp42Header, ipv4Header).WithFrameSize(512).WithFrameRatePct(5)
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+	return []*ondatra.Flow{ipv4FlowVlan10, ipv6FlowVlan10, ipipFlowVlan10, ipipDscp46FlowVlan10, ipipDscp42FlowVlan10,
+		ipv4FlowVlan20, ipv6FlowVlan20, ipipFlowVlan20, ipipDscp46FlowVlan20, ipipDscp42FlowVlan20}
+}
+
+func sendTraffic(t *testing.T, ate *ondatra.ATEDevice, allFlows []*ondatra.Flow) {
+	t.Logf("*** Sending traffic from ATE ...")
+	t.Logf("*** Starting traffic ...")
+	ate.Traffic().Start(t, allFlows...)
+	time.Sleep(trafficDuration)
+	t.Logf("*** Stop traffic ...")
+	ate.Traffic().Stop(t)
+}
+
+func captureTrafficStats(t *testing.T, ate *ondatra.ATEDevice, flowName string, wantLoss bool) {
+	afc := ate.Telemetry().Flow(flowName).Counters()
+	outPkts := afc.OutPkts().Get(t)
+	t.Logf("ate:Flow out counters %v %v", flowName, outPkts)
+	fptest.LogYgot(t, "ate:Flow counters", afc, afc.Get(t))
+
+	inPkts := afc.InPkts().Get(t)
+	t.Logf("ate:Flow in counters %v %v", flowName, inPkts)
+
+	lostPkts := inPkts - outPkts
+	t.Logf("Sent Packets: %d, received Packets: %d", outPkts, inPkts)
+	if !wantLoss {
+		if lostPkts > 0 {
+			t.Logf("Packets received not matching packets sent. Sent: %v, Received: %d", outPkts, inPkts)
+		} else {
+			t.Logf("Traffic Test Passed! Sent: %d, Received: %d", outPkts, inPkts)
+		}
+	}
+}
+
+func contains(item string, items []string) bool {
+	flag := false
+	for _, j := range items {
+		if item == j {
+			return true
+		}
+	}
+	return flag
+}
+
+// verifyTraffic confirms that every traffic flow has the expected amount of loss (0% or 100%
+func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, flows []*ondatra.Flow, passFlows []string) {
+	for _, flow := range flows {
+		t.Logf("*** Verifying %v traffic on ATE ... ", flow.Name())
+		captureTrafficStats(t, ate, flow.Name(), false)
+		lossPct := ate.Telemetry().Flow(flow.Name()).LossPct().Get(t)
+		if contains(flow.Name(), passFlows) {
+			if lossPct > 0 {
+				t.Errorf("Traffic Loss Pct for Flow: %s\n got %v, want 0", flow.Name(), lossPct)
```

Comment:
There should not be newlines in error messages.
`t.Errorf("Traffic Loss Pct for flow %s: got %v, want 0", flow.Name(), lossPct)`

Reply:
Done, in new commit. Thank You.